test/monitoring_console/monitoring_console_test.go

```diff
@@ -0,0 +1,513 @@
+				found[podName] = false
+				for _, peer := range peerList {
+					testenvInstance.Log.Info("Check Peer matches standalone pod", "Standalone Pod", podName, "Peer in peer list", peer)
+					if strings.Contains(peer, podName) {
+						found[podName] = true
+						break
+					}
+				}
+			}
+			allStandaloneConfigured := true
+			for _, key := range found {
+				if !key {
+					allStandaloneConfigured = false
+					break
+				}
+			}
+			Expect(allStandaloneConfigured).To(Equal(true))
+		})
+	})
+
+	Context("SearchHeadCluster deployment with Scale Up", func() {
+		It("MC can configure SHC instances in a namespace", func() {
+
+			_, err := deployment.DeploySearchHeadCluster(deployment.GetName(), "", "", "")
+			Expect(err).To(Succeed(), "Unable to deploy search head cluster")
+			// Ensure search head cluster go to Ready phase
+			shc := &enterprisev1.SearchHeadCluster{}
+			Eventually(func() splcommon.Phase {
+				err := deployment.GetInstance(deployment.GetName(), shc)
+				if err != nil {
+					return splcommon.PhaseError
+				}
+				testenvInstance.Log.Info("Waiting for search head cluster instance status to be ready", "instance", shc.ObjectMeta.Name, "Phase", shc.Status.Phase)
+				dumpGetPods(testenvInstance.GetName())
+				return shc.Status.Phase
+			}, deployment.GetTimeout(), PollInterval).Should(Equal(splcommon.PhaseReady))
+
+			// In a steady state, we should stay in Ready and not flip-flop around
+			Consistently(func() splcommon.Phase {
+				_ = deployment.GetInstance(deployment.GetName(), shc)
+				return shc.Status.Phase
+			}, ConsistentDuration, ConsistentPollInterval).Should(Equal(splcommon.PhaseReady))
+
+			// Check Monitoring Console Pod is in Ready State
+			Eventually(func() bool {
+				check := testenv.CheckMCPodReady(testenvInstance.GetName())
+				return check
+			}, deployment.GetTimeout(), PollInterval).Should(Equal(true))
+
+			// Check Monitoring console is configured with all search head instances in namespace
+			peerList := testenv.GetConfiguredPeers(testenvInstance.GetName())
+			found := make(map[string]bool)
+			testenvInstance.Log.Info("Peer List", "instance", peerList)
+			for i := 0; i < 3; i++ {
+				podName := "splunk-" + deployment.GetName() + "-search-head-" + strconv.Itoa(i)
+				found[podName] = false
+				for _, peer := range peerList {
+					testenvInstance.Log.Info("Check Peer matches serach head pod", "Search Head Pod", podName, "Peer in peer list", peer)
+					if strings.Contains(peer, podName) {
+						found[podName] = true
+						break
+					}
+				}
+			}
+			allSearchHeadsConfigured := true
+			for _, key := range found {
+				if !key {
+					allSearchHeadsConfigured = false
+					break
+				}
+			}
+			Expect(allSearchHeadsConfigured).To(Equal(true))
+
+			// Scale Search Head Cluster
+			testenvInstance.Log.Info("Scaling search head cluster")
+			_, err = exec.Command("kubectl", "scale", "shc", "-n", testenvInstance.GetName(), deployment.GetName(), "--replicas=4").Output()
```

Comment:
do we need to handle error here?

Reply:
Yes there is no point in proceeding with test if we are not able to scale the SHC.